### ISSUE TITLE
Problem with error handling on create user?

### ISSUE THREAD

**Issue by CoreyJ87** [sqmk/huejay#59]
Im new to node. So  I may just be doing something wrong. Although, when I run create user. I get this error back which is good.

Error: Huejay: Philips Hue: 101, link button not pressed
    at HuejayError (/Users/corey/repos/hueadmin/node_modules/huejay/lib/Error.js:21:5)
    at getHttpClient.request.then.catch.error (/Users/corey/repos/hueadmin/node_modules/huejay/lib/Transport.js:102:15)
    at process._tickCallback (internal/process/next_tick.js:103:7)

Although for some reason this code below does not work for me. It never triggers that if statement because  error.type is undefined

  if (error instanceof huejay.Error && error.type === 101) {
        res.send(`Link button not pressed. Try again...`)
        return console.log(`Link button not pressed. Try again...`);
      }
      console.log(error.stack);

I would like to catch it correctly so I can output what I would like to the UI. Instead of returning the entire error message. Thanks for writing this library though. Its great.

**Comment by sqmk** [comment]
Hi Corey,

I'll take a look and see if I can reproduce locally.

**Comment by sqmk** [comment]
Can you show me a bit more of your usage? The combination of Huejay: and Philips Hue: in the error string seems fairly suspect.

**Comment by CoreyJ87** [comment]
createUser(deviceIP){
    console.log('Attempting to create user....');
    console.log('Make sure link button on bridge is pressed.....');
    
    let client = new huejay.Client({
      host:     deviceIP,
      username: 'default'  //'faDU2YmXESWEGdLea3kkZi-8CB7tSuhuPl6ZDBob'
    });
    console.log(client);
    
    let user = new client.users.User;
    user.deviceType = 'customDevice';

    return client.users.create(user)
    .then(user => {
      console.log(`New user: ${user.username} created. With device type: ${user.deviceType}`);
      //Write user to creds file
      fs.writeFile("./creds", user.username, function(err) {
        if(err) {
          return console.log(err);
        }
      });
    })
    .catch(error => {
      if (error instanceof huejay.Error && error.type === 101) {
        res.send(`Link button not pressed. Try again...`)
        return console.log(`Link button not pressed. Try again...`);
      }
      console.log(error.stack);
    })
    .catch(error => {
      console.log(error.message);
    });
  }

So i have this inside a class and i'm returning the promise to my app.js if the button is pressed and the user creation succeeds. Although, if I dont press the button. I get what I posted above.

**Comment by CoreyJ87** [comment]
Alil update. I just accidentally sent an undefined value to a bulb. Got this error. Its doing the same thing Error: Huejay: Philips Hue: 7, invalid value, null,, for parameter, bri
    at HuejayError (/Users/corey/repos/hueadmin/node_modules/huejay/lib/Error.js:21:5)
    at getHttpClient.request.then.catch.error (/Users/corey/repos/hueadmin/node_modules/huejay/lib/Transport.js:102:15)
    at process._tickCallback (internal/process/next_tick.js:103:7)

Could this be because of my node version or something?

**Comment by sqmk** [comment]
Sorry, didn't get a chance to look at this yesterday due to network outage in most of Texas (AT&T :( ).

Looking at this now.

**Comment by vtabary** [comment]
Hi,

I got the same error. I don't think it's about the library. In fact, if I do the same request but with an xhr request from http://<bridge ip address>/debug/clip.html, I get the same answer.

To check this, I just followed the get started guide : http://www.developers.meethue.com/documentation/getting-started

I don't have a solution yet ...

**Comment by vtabary** [comment]
Well, I do now (yes, that was quick).

Push the link button before making the request. The request to create the user is not synchronous, so the library (or our app) should check something every second if the button has been pushed.

**Comment by CoreyJ87** [comment]
@vtabary, I mentioned above that it does work when you actually press the button. Although, the point of that error message is to be able to catch it and handle it properly. Currently, I just spit out a message saying that the button isnt pressed. Although, what if for some reason there is another error? Or you want to catch that error and handle it different than all errors? Ideally you would want to be able to actually check the error code and whatnot.

**Comment by vtabary** [comment]
Yes, you're right, I'm sorry, it was late ...
But there is a good point, I confirmed all what you said, even the "type" property which is undefined.

I think the problem is here : https://github.com/sqmk/huejay/blob/master/lib/Transport.js#L102

The "throw HuejayError" in the "then" is caught by the "catch" next
I can't send a pull request for now but It should be fixed by adding the type to the HuejayError, something like : 

```
      .catch(error => {
        throw new HuejayError({
          message: error.message,
          type: error.type
        });
      });
```

**Comment by sqmk** [comment]
This is now fixed.

Released via NPM as version 1.1.5!
